acceptance/tests/pxp-module-puppet/restart_pxp_agent_during_non_blocking_run.rb

```diff
@@ -0,0 +1,133 @@
+require 'pxp-agent/test_helper.rb'
+require 'json'
+
+ENVIRONMENT_NAME = 'sleep'
+MODULE_NAME = 'sleep'
+SECONDS_TO_SLEEP = 500 # The test will use SIGALARM to end this as soon as required
+STATUS_QUERY_MAX_RETRIES = 60
+STATUS_QUERY_INTERVAL_SECONDS = 5
+
+test_name 'C94705 - Run Puppet (non-blocking request) and restart pxp-agent service during run' do
+
+  applicable_agents = agents.select { |agent| agent['platform'] !~ /win/}
+  if applicable_agents.empty? then
+    skip_test('Skipping - All agent hosts are Windows and are not applicable to this test (see PCP-276)')
+  end
+
+  teardown do
+    if (!applicable_agents.empty?)
+      applicable_agents.each do |agent|
+        # If puppet agent run has been left running by this test failing, terminate it
+        if agent.file_exist?('/opt/puppetlabs/puppet/cache/state/agent_catalog_run.lock')
+          on(agent 'kill -9 `cat /opt/puppetlabs/puppet/cache/state/agent_catalog_run.lock`')
+        end
+      end
+    end
+  end
+
+  step 'On master, create a new environment with a module that will take some time' do
+    environmentpath = master.puppet['environmentpath']
+    site_manifest = "#{environmentpath}/#{ENVIRONMENT_NAME}/manifests/site.pp"
+    on(master, "cp -r #{environmentpath}/production #{environmentpath}/#{ENVIRONMENT_NAME}")
```

Comment:
note, i have one solution for this in PR-405.
https://github.com/puppetlabs/pxp-agent/pull/405